pkg/csi_driver/reconciler.go

```diff
@@ -165,14 +174,343 @@ func (recon *multishareReconciler) reconcileWorker() {
+				klog.Infof("Starting share Delete operation for %s", shareURI)
+				_, err = recon.cloud.File.StartDeleteShareOp(context.TODO(), share)
+			} else if shareInfo.Status.ShareStatus != v1alpha1.READY {
+				klog.Infof("Starting share Create operation for %s", shareURI)
+				_, err = recon.cloud.File.StartCreateShareOp(context.TODO(), share)
+			} else if shareInfo.Status.CapacityBytes != 0 && shareInfo.Spec.CapacityBytes != shareInfo.Status.CapacityBytes {
+				klog.Infof("Starting share Resize operation for %s", shareURI)
+				_, err = recon.cloud.File.StartResizeShareOp(context.TODO(), share)
+			}
+		}
+		if err != nil {
+			recon.updateShareInfoErr(shareInfo, err)
+		}
+	}
+}
+
+func (recon *MultishareReconciler) maybeMarkShareInfoStatusDeleted(shareInfo *v1alpha1.ShareInfo, instanceInfo *v1alpha1.InstanceInfo) (*v1alpha1.InstanceInfo, error) {
+	if instanceInfo == nil {
+		return instanceInfo, fmt.Errorf("instanceInfo does not exist for shareInfo %s which is assigned to %q", shareInfo.Name, shareInfo.Status.InstanceHandle)
+	}
+
+	klog.Infof("mark shareInfo %s ShareStatus as deleted", shareInfo.Name)
+	instanceInfoClone := instanceInfo.DeepCopy()
+	instanceInfoClone, updated := recon.removeShareFromInstanceInfo(instanceInfoClone, shareInfo.Name)
+	if updated {
+		klog.Infof("trying to remove share %s from assigned instanceInfo %s", shareInfo.Name, instanceInfo.Name)
+		var err error
+		instanceInfoClone, err = recon.updateInstanceInfoStatus(context.TODO(), instanceInfoClone)
+		if err != nil {
+			return instanceInfo, fmt.Errorf("failed to remove share form instanceInfo Status: %s", err.Error())
+		}
+	}
+	shareInfoClone := shareInfo.DeepCopy()
+	if shareInfo.Status == nil {
+		shareInfoClone.Status = &v1alpha1.ShareInfoStatus{}
+	}
+	shareInfoClone.Status.ShareStatus = v1alpha1.DELETED
+	_, err := recon.updateShareInfoStatus(context.TODO(), shareInfoClone)
+	if err != nil {
+		return instanceInfoClone, fmt.Errorf("failed to update %s.Status.ShareStatus to DELETED: %s", shareInfo.Name, err.Error())
+	}
+
+	return instanceInfoClone, nil
+}
+
+func (recon *MultishareReconciler) sendInstanceRequests(instanceInfos map[string]*v1alpha1.InstanceInfo, ops []*Op) {
+	for _, instanceInfo := range instanceInfos {
+		needDelete := instanceInfo.DeletionTimestamp != nil
+		if !needDelete && instanceInfo.Status != nil &&
+			instanceInfo.Spec.CapacityBytes == instanceInfo.Status.CapacityBytes && instanceInfo.Status.InstanceStatus == v1alpha1.READY {
+			klog.V(6).Infof("no need to send any instance request for %s", instanceInfo.Name)
+			continue
+		}
+
+		instanceURI := util.InstanceInfoNameToInstanceURI(instanceInfo.Name)
+		op, err := runningOpMaybeErrForTarget(instanceURI, ops)
+		if err != nil {
+			recon.updateInstanceInfoErr(instanceInfo, err)
+		}
+		if op == nil {
+			klog.Infof("no running Op found for %s", instanceURI)
+			var instance *file.MultishareInstance
+			instance, err = recon.generateNewMultishareInstance(instanceInfo)
```

Comment:
Is this needed for operations other than instance creation? It reserves Ip range inside the function which is not needed for delete and resize instance operations

Similarlly we then dont need releaseIpRange in L364 for all operation types?

Reply:
in L489 we'll only reserve Ip range if the instance does not already have cidr AND sc requires ip reservation. I think it makes it easier to have a unified logic to fill in all the instance info before sending it out to filestore API